modules/pam_nologin/pam_nologin.c

```diff
@@ -99,21 +99,25 @@ static int perform_check(pam_handle_t *pamh, struct opt_s *opts)
 	    goto clean_up_fd;
 	}
 
-	mtmp = malloc(st.st_size+1);
-	if (!mtmp) {
-	    pam_syslog(pamh, LOG_CRIT, "out of memory");
-	    retval = PAM_BUF_ERR;
-	    goto clean_up_fd;
-	}
-
-	if (pam_modutil_read(fd, mtmp, st.st_size) == st.st_size) {
-		mtmp[st.st_size] = '\0';
-		(void) pam_prompt (pamh, msg_style, NULL, "%s", mtmp);
+	/* Don't print anything if the message is empty, will only
+	   disturb the output with empty lines */
+	if (st.st_size > 0) {
+	    mtmp = malloc(st.st_size+1);
```

Comment:
Could you also move the declaration of `mtmp` from
https://github.com/linux-pam/linux-pam/blob/016d5e2bc0423f5226191143329ad7e50fde82fd/modules/pam_nologin/pam_nologin.c#L82
inside this new block, e.g.
```
if (st.st_size > 0) {
    char *mtmp = malloc(st.st_size+1);
```
?